instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/job/JobBuilderHelperInstrumentation.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.job;
+
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isProtected;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.javaagent.instrumentation.api.ContextStore;
+import io.opentelemetry.javaagent.instrumentation.api.InstrumentationContext;
+import io.opentelemetry.javaagent.instrumentation.spring.batch.ContextAndScope;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.job.builder.JobBuilderHelper;
+
+public class JobBuilderHelperInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<? super TypeDescription> typeMatcher() {
+    // Java DSL Job config
+    return named("org.springframework.batch.core.job.builder.JobBuilderHelper");
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        named("enhance")
```

Comment:
Do you want to instrument enhance instead of the exit of the constructor?

Reply:
Hmm, I don't really have a hard preference, but `enhance()` still seems like a proper instrumentation point for this. Is there any advantage of instrumenting the constructor that I might not be aware of?

Reply:
I think the main point was it seems like enhance can be called multiple times, so constructor makes it easier to prevent dedupe. Also not sure if we want our listener to be in the beginning or end - if end is good, then enhance is better. I think we are able to pick which one here.

Reply:
Ordering is defined by the `Ordered` interface which all listeners in this PR implement - in general, we want them as soon as possible. It doesn't really matter when we add the listener to the job/step, its order property is much more significant.
No duplication of tracing listeners is guaranteed by `equals()` & `hashCode()` methods.